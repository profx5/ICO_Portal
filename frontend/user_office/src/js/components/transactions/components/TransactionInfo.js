import React, {Component} from 'react'
import styled from 'styled-components';

import iconCheckGreen from './../../../../img/check-green.svg';


const TransactionsInfo = ({transferHashLink, payementHashLink, amount, currency, rate_usdc, usdc_value, bonus_percent, tokens}) => {
    
    const rateUSD = (rate_usdc / 100).toFixed(2),
        USDValue = usdc_value.toFixed(2),
        baseTokens = (USDValue / 2).toFixed(2),
        bonusTokens = (USDValue / 2 * bonus_percent / 100).toFixed(2)

    return (
        <Wrapper>
            <InfoBlock>
                <div className="TransactionContent_head">Blockchain transactions</div>
                <div>
                    <div className="TransactionContent_info">
                        <div className="TransactionContent_propertyName">Payment:</div>
                        <div className="TransactionContent_propertyLink"><a target="_blank" href={payementHashLink}>{payementHashLink}</a></div>
                        <div className="TransactionContent_propertyIconWrapper"><img alt="" src={iconCheckGreen}/></div>
                    </div>
                    {transferHashLink && 
                        <div className="TransactionContent_info">
                            <div className="TransactionContent_propertyName">Transfer:</div>
                            <div className="TransactionContent_propertyLink"><a target="_blank" href={transferHashLink}>{transferHashLink}</a></div>
                            <div className="TransactionContent_propertyIconWrapper"><img alt="" src={iconCheckGreen}/></div>
                        </div>
                    }
                </div>
            </InfoBlock>
            
            <InfoBlock>
                <Part className="Info_part-1">
                    <div className="Info_head">Calculated token amount</div>
                    <div className="Info_partContent">

                        <div>{`${amount} ${currency} x ` + rateUSD + " USD = " + USDValue + " USD"}<br/></div>
                        <div>{"Phase bonus = " + bonus_percent + "%"}<br/></div>
                        <div>Token base price = 2 USD <br/></div>
                        <div>{"Base tokens: " + USDValue + " USD / 2 = " + baseTokens + " OGD"}<br/></div>
                        <div>{"Bonus tokens: " + baseTokens + " x " + bonus_percent + "% = " + bonusTokens + " OGD"}<br/></div>
                        <div>{"Total: " + tokens}<br/></div>

                    </div>
                </Part>
                <Part className="Info_part-2">
                    <div className="Info_head">Referral bonuses</div>
                    <div className="Info_partContent">—</div>
                </Part>
                <Part className="Info_part-3">
                    <div className="Info_head">Withdrawal</div>
                    <div className="Info_partContent">—</div>
                </Part>
            </InfoBlock>
        </Wrapper>
    )
}


export default TransactionsInfo;

const Wrapper = styled.div`
    flex-basis: 100%;
    border-left: 2px solid rgb(67, 129, 252);
    padding-left: 30px;
    display: flex;
    flex-flow: row wrap;
`;

const InfoBlock = styled.div`
    flex-basis: 100%;
    &:first-of-type {

    }
    &:last-of-type {
        display: flex;
        border-top: 1px solid rgba(151, 151, 151, .2);
        border-bottom: 1px solid rgba(151, 151, 151, .2);
        padding: 35px 0 10px;
        margin-top: 20px;
        margin-bottom: 35px;
    }
    .TransactionContent_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }

    .TransactionContent_info {
        flex-basis: 100%;
        display: flex;
        flex-flow: row nowrap;
        justify-content: space-between;
        margin-bottom: 10px;
    }
    .TransactionContent_propertyName {
        min-width: 80px;
    }
    .TransactionContent_propertyLink {
        flex: 1;
    }
    .TransactionContent_propertyIconWrapper {
        width: 35px;
        img {
            width: 22px;
            height: auto;
            float: right;
        }
    }
`;

const Part = styled.div`
    flex-basis: 33.33333%;
    &.Info_part-1 {
        padding-right: 15px;
        .Info_partContent {
            div {
                margin-bottom: 10px;
            }
        }
    }
    &.Info_part-2 {
        border-left: 1px solid rgba(151, 151, 151, .2);
        border-right: 1px solid rgba(151, 151, 151, .2);
    }
    &.Info_part-3 {

    }
    &.Info_part-2, &.Info_part-3 {
        position: relative;
        .Info_head {
            padding-left: 35px;
        }
        .Info_partContent {
            position: absolute;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            display: flex;
            justify-content: center;
            align-items: center;
        }
    }
    .Info_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }
    .Info_head ~ div {
        margin-bottom: 10px;
    }
`;
