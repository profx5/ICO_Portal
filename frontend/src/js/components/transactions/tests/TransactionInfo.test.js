import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TransactionInfo from 'js/components/transactions/stateless/TransactionInfo';


configure({ adapter: new Adapter() });

it('should render stateless TransactionInfo', () => {
    const component = shallow(<TransactionInfo/>);
    expect(component).toMatchSnapshot();
})
