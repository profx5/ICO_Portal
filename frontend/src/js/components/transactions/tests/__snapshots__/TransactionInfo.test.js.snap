// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless TransactionInfo 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransactionsInfo
    usdc_value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <div
            className="TransactionContent_head"
          >
            Blockchain transactions
          </div>
          <div>
            <div
              className="TransactionContent_info"
            >
              <div
                className="TransactionContent_propertyName"
              >
                Payment:
              </div>
              <div
                className="TransactionContent_propertyLink"
              >
                <a
                  target="_blank"
                />
              </div>
              <div
                className="TransactionContent_propertyIconWrapper"
              >
                <img
                  alt=""
                  src="check-green.svg"
                />
              </div>
            </div>
          </div>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef
            className="Info_part-1"
          >
            <div
              className="Info_head"
            >
              Calculated token amount
            </div>
            <div
              className="Info_partContent"
            >
              <div>
                undefined undefined x NaN USD = 0.00 USD
                <br />
              </div>
              <div>
                Token price = undefined USD
                <br />
              </div>
              <div>
                Total: undefined
                <br />
              </div>
            </div>
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="TransactionContent_head"
            >
              Blockchain transactions
            </div>,
            <div>
              <div
                className="TransactionContent_info"
              >
                <div
                  className="TransactionContent_propertyName"
                >
                  Payment:
                </div>
                <div
                  className="TransactionContent_propertyLink"
                >
                  <a
                    target="_blank"
                  />
                </div>
                <div
                  className="TransactionContent_propertyIconWrapper"
                >
                  <img
                    alt=""
                    src="check-green.svg"
                  />
                </div>
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Blockchain transactions",
              "className": "TransactionContent_head",
            },
            "ref": null,
            "rendered": "Blockchain transactions",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="TransactionContent_info"
                >
                  <div
                    className="TransactionContent_propertyName"
                  >
                    Payment:
                  </div>
                  <div
                    className="TransactionContent_propertyLink"
                  >
                    <a
                      target="_blank"
                    />
                  </div>
                  <div
                    className="TransactionContent_propertyIconWrapper"
                  >
                    <img
                      alt=""
                      src="check-green.svg"
                    />
                  </div>
                </div>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="TransactionContent_propertyName"
                    >
                      Payment:
                    </div>,
                    <div
                      className="TransactionContent_propertyLink"
                    >
                      <a
                        target="_blank"
                      />
                    </div>,
                    <div
                      className="TransactionContent_propertyIconWrapper"
                    >
                      <img
                        alt=""
                        src="check-green.svg"
                      />
                    </div>,
                  ],
                  "className": "TransactionContent_info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Payment:",
                      "className": "TransactionContent_propertyName",
                    },
                    "ref": null,
                    "rendered": "Payment:",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        target="_blank"
                      />,
                      "className": "TransactionContent_propertyLink",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "href": undefined,
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        alt=""
                        src="check-green.svg"
                      />,
                      "className": "TransactionContent_propertyIconWrapper",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "",
                        "src": "check-green.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    flex-basis: 100%;
    &:last-of-type {
        display: flex;
        border-top: 1px solid rgba(151, 151, 151, .2);
        border-bottom: 1px solid rgba(151, 151, 151, .2);
        padding: 35px 0 10px;
        margin-top: 20px;
        margin-bottom: 35px;
    }
    .TransactionContent_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }

    .TransactionContent_info {
        flex-basis: 100%;
        display: flex;
        flex-flow: row nowrap;
        justify-content: space-between;
        margin-bottom: 10px;
    }
    .TransactionContent_propertyName {
        min-width: 80px;
    }
    .TransactionContent_propertyLink {
        flex: 1;
    }
    .TransactionContent_propertyIconWrapper {
        width: 35px;
        img {
            width: 22px;
            height: auto;
            float: right;
        }
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            className="Info_part-1"
          >
            <div
              className="Info_head"
            >
              Calculated token amount
            </div>
            <div
              className="Info_partContent"
            >
              <div>
                undefined undefined x NaN USD = 0.00 USD
                <br />
              </div>
              <div>
                Token price = undefined USD
                <br />
              </div>
              <div>
                Total: undefined
                <br />
              </div>
            </div>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="Info_head"
              >
                Calculated token amount
              </div>,
              <div
                className="Info_partContent"
              >
                <div>
                  undefined undefined x NaN USD = 0.00 USD
                  <br />
                </div>
                <div>
                  Token price = undefined USD
                  <br />
                </div>
                <div>
                  Total: undefined
                  <br />
                </div>
              </div>,
            ],
            "className": "Info_part-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Calculated token amount",
                "className": "Info_head",
              },
              "ref": null,
              "rendered": "Calculated token amount",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    undefined undefined x NaN USD = 0.00 USD
                    <br />
                  </div>,
                  <div>
                    Token price = undefined USD
                    <br />
                  </div>,
                  <div>
                    Total: undefined
                    <br />
                  </div>,
                ],
                "className": "Info_partContent",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "undefined undefined x NaN USD = 0.00 USD",
                      <br />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "undefined undefined x NaN USD = 0.00 USD",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Token price = undefined USD",
                      <br />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Token price = undefined USD",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Total: undefined",
                      <br />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Total: undefined",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    flex-basis: 33.33333%;
    &.Info_part-1 {
        padding-right: 15px;
        .Info_partContent {
            div {
                margin-bottom: 10px;
                ",
                "@media (max-width: 767px)",
                " {
                    margin-bottom: 5px;
                }
            }
        }
    }
    &.Info_part-2 {
        border-left: 1px solid rgba(151, 151, 151, .2);
        border-right: 1px solid rgba(151, 151, 151, .2);
    }
    &.Info_part-2, &.Info_part-3 {
        position: relative;
        .Info_head {
            padding-left: 35px;
        }
        .Info_partContent {
            position: absolute;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            display: flex;
            justify-content: center;
            align-items: center;
        }
    }
    .Info_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }
    .Info_head ~ div {
        margin-bottom: 10px;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    flex-basis: 100%;
    &:last-of-type {
        display: flex;
        border-top: 1px solid rgba(151, 151, 151, .2);
        border-bottom: 1px solid rgba(151, 151, 151, .2);
        padding: 35px 0 10px;
        margin-top: 20px;
        margin-bottom: 35px;
    }
    .TransactionContent_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }

    .TransactionContent_info {
        flex-basis: 100%;
        display: flex;
        flex-flow: row nowrap;
        justify-content: space-between;
        margin-bottom: 10px;
    }
    .TransactionContent_propertyName {
        min-width: 80px;
    }
    .TransactionContent_propertyLink {
        flex: 1;
    }
    .TransactionContent_propertyIconWrapper {
        width: 35px;
        img {
            width: 22px;
            height: auto;
            float: right;
        }
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    flex-basis: 100%;
    border-left: 2px solid rgb(67, 129, 252);
    padding-left: 30px;
    display: flex;
    flex-flow: row wrap;
    ",
          "@media (max-width: 767px)",
          " {
        padding-left: 14px;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <div
              className="TransactionContent_head"
            >
              Blockchain transactions
            </div>
            <div>
              <div
                className="TransactionContent_info"
              >
                <div
                  className="TransactionContent_propertyName"
                >
                  Payment:
                </div>
                <div
                  className="TransactionContent_propertyLink"
                >
                  <a
                    target="_blank"
                  />
                </div>
                <div
                  className="TransactionContent_propertyIconWrapper"
                >
                  <img
                    alt=""
                    src="check-green.svg"
                  />
                </div>
              </div>
            </div>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              className="Info_part-1"
            >
              <div
                className="Info_head"
              >
                Calculated token amount
              </div>
              <div
                className="Info_partContent"
              >
                <div>
                  undefined undefined x NaN USD = 0.00 USD
                  <br />
                </div>
                <div>
                  Token price = undefined USD
                  <br />
                </div>
                <div>
                  Total: undefined
                  <br />
                </div>
              </div>
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="TransactionContent_head"
              >
                Blockchain transactions
              </div>,
              <div>
                <div
                  className="TransactionContent_info"
                >
                  <div
                    className="TransactionContent_propertyName"
                  >
                    Payment:
                  </div>
                  <div
                    className="TransactionContent_propertyLink"
                  >
                    <a
                      target="_blank"
                    />
                  </div>
                  <div
                    className="TransactionContent_propertyIconWrapper"
                  >
                    <img
                      alt=""
                      src="check-green.svg"
                    />
                  </div>
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Blockchain transactions",
                "className": "TransactionContent_head",
              },
              "ref": null,
              "rendered": "Blockchain transactions",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="TransactionContent_info"
                  >
                    <div
                      className="TransactionContent_propertyName"
                    >
                      Payment:
                    </div>
                    <div
                      className="TransactionContent_propertyLink"
                    >
                      <a
                        target="_blank"
                      />
                    </div>
                    <div
                      className="TransactionContent_propertyIconWrapper"
                    >
                      <img
                        alt=""
                        src="check-green.svg"
                      />
                    </div>
                  </div>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="TransactionContent_propertyName"
                      >
                        Payment:
                      </div>,
                      <div
                        className="TransactionContent_propertyLink"
                      >
                        <a
                          target="_blank"
                        />
                      </div>,
                      <div
                        className="TransactionContent_propertyIconWrapper"
                      >
                        <img
                          alt=""
                          src="check-green.svg"
                        />
                      </div>,
                    ],
                    "className": "TransactionContent_info",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Payment:",
                        "className": "TransactionContent_propertyName",
                      },
                      "ref": null,
                      "rendered": "Payment:",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          target="_blank"
                        />,
                        "className": "TransactionContent_propertyLink",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                          "href": undefined,
                          "target": "_blank",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "a",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <img
                          alt=""
                          src="check-green.svg"
                        />,
                        "className": "TransactionContent_propertyIconWrapper",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "alt": "",
                          "src": "check-green.svg",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    flex-basis: 100%;
    &:last-of-type {
        display: flex;
        border-top: 1px solid rgba(151, 151, 151, .2);
        border-bottom: 1px solid rgba(151, 151, 151, .2);
        padding: 35px 0 10px;
        margin-top: 20px;
        margin-bottom: 35px;
    }
    .TransactionContent_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }

    .TransactionContent_info {
        flex-basis: 100%;
        display: flex;
        flex-flow: row nowrap;
        justify-content: space-between;
        margin-bottom: 10px;
    }
    .TransactionContent_propertyName {
        min-width: 80px;
    }
    .TransactionContent_propertyLink {
        flex: 1;
    }
    .TransactionContent_propertyIconWrapper {
        width: 35px;
        img {
            width: 22px;
            height: auto;
            float: right;
        }
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              className="Info_part-1"
            >
              <div
                className="Info_head"
              >
                Calculated token amount
              </div>
              <div
                className="Info_partContent"
              >
                <div>
                  undefined undefined x NaN USD = 0.00 USD
                  <br />
                </div>
                <div>
                  Token price = undefined USD
                  <br />
                </div>
                <div>
                  Total: undefined
                  <br />
                </div>
              </div>
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div
                  className="Info_head"
                >
                  Calculated token amount
                </div>,
                <div
                  className="Info_partContent"
                >
                  <div>
                    undefined undefined x NaN USD = 0.00 USD
                    <br />
                  </div>
                  <div>
                    Token price = undefined USD
                    <br />
                  </div>
                  <div>
                    Total: undefined
                    <br />
                  </div>
                </div>,
              ],
              "className": "Info_part-1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Calculated token amount",
                  "className": "Info_head",
                },
                "ref": null,
                "rendered": "Calculated token amount",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      undefined undefined x NaN USD = 0.00 USD
                      <br />
                    </div>,
                    <div>
                      Token price = undefined USD
                      <br />
                    </div>,
                    <div>
                      Total: undefined
                      <br />
                    </div>,
                  ],
                  "className": "Info_partContent",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "undefined undefined x NaN USD = 0.00 USD",
                        <br />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "undefined undefined x NaN USD = 0.00 USD",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Token price = undefined USD",
                        <br />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Token price = undefined USD",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Total: undefined",
                        <br />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Total: undefined",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
    flex-basis: 33.33333%;
    &.Info_part-1 {
        padding-right: 15px;
        .Info_partContent {
            div {
                margin-bottom: 10px;
                ",
                  "@media (max-width: 767px)",
                  " {
                    margin-bottom: 5px;
                }
            }
        }
    }
    &.Info_part-2 {
        border-left: 1px solid rgba(151, 151, 151, .2);
        border-right: 1px solid rgba(151, 151, 151, .2);
    }
    &.Info_part-2, &.Info_part-3 {
        position: relative;
        .Info_head {
            padding-left: 35px;
        }
        .Info_partContent {
            position: absolute;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            display: flex;
            justify-content: center;
            align-items: center;
        }
    }
    .Info_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }
    .Info_head ~ div {
        margin-bottom: 10px;
    }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    flex-basis: 100%;
    &:last-of-type {
        display: flex;
        border-top: 1px solid rgba(151, 151, 151, .2);
        border-bottom: 1px solid rgba(151, 151, 151, .2);
        padding: 35px 0 10px;
        margin-top: 20px;
        margin-bottom: 35px;
    }
    .TransactionContent_head {
        font-weight: 600;
        margin-bottom: 20px;
        font-size: 18px;
    }

    .TransactionContent_info {
        flex-basis: 100%;
        display: flex;
        flex-flow: row nowrap;
        justify-content: space-between;
        margin-bottom: 10px;
    }
    .TransactionContent_propertyName {
        min-width: 80px;
    }
    .TransactionContent_propertyLink {
        flex: 1;
    }
    .TransactionContent_propertyIconWrapper {
        width: 35px;
        img {
            width: 22px;
            height: auto;
            float: right;
        }
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    flex-basis: 100%;
    border-left: 2px solid rgb(67, 129, 252);
    padding-left: 30px;
    display: flex;
    flex-flow: row wrap;
    ",
            "@media (max-width: 767px)",
            " {
        padding-left: 14px;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
